### PR TITLE
Fix download artifacts message

### PR DESCRIPTION
**Pull request by omeshp**

### CODE REVIEW

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 1]
```diff
@@ -221,15 +245,18 @@ private string GetCommitsWorkFolder(IExecutionContext context)
...
                     async () =>
-                        {
-                            var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
-                            executionContext.Output(StringUtil.Loc("RMEnsureArtifactFolderExistsAndIsClean", downloadFolderPath));
-                            releaseFileSystemManager.EnsureEmptyDirectory(downloadFolderPath, executionContext.CancellationToken);
+                    {
+                        var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
+                        executionContext.Output(StringUtil.Loc("RMEnsureArtifactFolderExistsAndIsClean",
+                            downloadFolderPath));
+                        releaseFileSystemManager.EnsureEmptyDirectory(downloadFolderPath,
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: shouldn't this happen in Initialize with this refactoring? the download artifact tasks would expect the folder to be empty, so agent has to empty it in initialize itself as download as part of agent might not happen?

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 2]
```diff
@@ -109,7 +116,22 @@ public override void ConvertLocalPath(IExecutionContext context, string localPat
 
             try
             {
-                ServiceEndpoint vssEndpoint = executionContext.Endpoints.FirstOrDefault(e => string.Equals(e.Name, ServiceEndpoints.SystemVssConnection, StringComparison.OrdinalIgnoreCase));
+                await DownloadArtifacts(executionContext, ReleaseArtifacts, ArtifactsWorkingFolder);
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: rename GetArtifactsAsync to DownloadArtifactsAndCommitsAsync?

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 3]
```diff
@@ -320,6 +347,15 @@ public override void InitializeJobExtension(IExecutionContext executionContext)
                 executionContext.Output(StringUtil.Loc("RMUserChoseToSkipArtifactDownload"));
                 Trace.Info("Skipping artifact download based on the setting specified.");
             }
+            else
+            {
+                ReleaseArtifacts = GetReleaseArtifacts(executionContext);
+
+                if (!ReleaseArtifacts.Any())
+                {
+                    Trace.Info("No artifacts found to be downloaded by agent.");
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: log this also in task logs (and so load message from resource)?

Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: within if also, CreateWorkingFolderIfRequired should be called?

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 4]
```diff
@@ -173,31 +187,41 @@ private string GetCommitsWorkFolder(IExecutionContext context)
...
+            executionContext.Output(StringUtil.Loc("RMDownloadingArtifact"));
+
             foreach (AgentArtifactDefinition agentArtifactDefinition in agentArtifactDefinitions)
             {
                 // We don't need to check if its old style artifact anymore. All the build data has been fixed and all the build artifact has Alias now.
                 ArgUtil.NotNullOrEmpty(agentArtifactDefinition.Alias, nameof(agentArtifactDefinition.Alias));
 
                 var extensionManager = HostContext.GetService<IExtensionManager>();
-                IArtifactExtension extension = (extensionManager.GetExtensions<IArtifactExtension>()).FirstOrDefault(x => agentArtifactDefinition.ArtifactType == x.ArtifactType);
+                IArtifactExtension extension =
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: breaking into multiple lines is causing more churn... would have been good if you had avoided making this change as part of these changes. is this triggered by VS code editor auto correcting line endings?

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 3 above)
Comment (omeshp) [reply src/Agent.Worker/Release/ReleaseJobExtension.cs]: This is just for info, there can be artifacts which will be downloaded using task. Printing this message will just confuse the user.

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 2 above)
Comment (omeshp) [reply src/Agent.Worker/Release/ReleaseJobExtension.cs]: resolved

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 4 above)
Comment (omeshp) [reply src/Agent.Worker/Release/ReleaseJobExtension.cs]: auto indented by vs.

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 1 above)
Comment (omeshp) [reply src/Agent.Worker/Release/ReleaseJobExtension.cs]: added CreateArtifactsFolder in initialize, if no artifacts to be downloaded by agent.

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 5]
```diff
@@ -320,7 +347,44 @@ public override void InitializeJobExtension(IExecutionContext executionContext)
...
+
+                if (!ReleaseArtifacts.Any())
+                {
+                    CreateArtifactsFolder(executionContext, ArtifactsWorkingFolder);
+                    Trace.Info("No artifacts found to be downloaded by agent.");
+                }
+            }
+
+            CheckForAvailableDiskSpace(executionContext);
+
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: remove blank line

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 6]
```diff
@@ -320,7 +347,44 @@ public override void InitializeJobExtension(IExecutionContext executionContext)
...
+        }
+
+        private void CheckForAvailableDiskSpace(IExecutionContext executionContext)
+        {
+            try
+            {
+                foreach (var drive in DriveInfo.GetDrives())
+                {
+                    var root = Path.GetPathRoot(ArtifactsWorkingFolder);
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: move this outside for loop

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 7]
```diff
@@ -320,7 +347,44 @@ public override void InitializeJobExtension(IExecutionContext executionContext)
...
 
+                    if (string.Equals(root, drive.Name, StringComparison.OrdinalIgnoreCase))
+                    {
+                        var availableSpaceInMB = drive.AvailableFreeSpace / (1024 * 1024);
+                        if (availableSpaceInMB < 100)
+                        {
+                            executionContext.Warning(StringUtil.Loc("RMLowAvailableDiskSpace", root));
+                        }
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: add break if drive match is found

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 8]
```diff
@@ -320,7 +347,44 @@ public override void InitializeJobExtension(IExecutionContext executionContext)
...
+
+            CheckForAvailableDiskSpace(executionContext);
+
+        }
+
+        private void CheckForAvailableDiskSpace(IExecutionContext executionContext)
+        {
+            try
+            {
+                foreach (var drive in DriveInfo.GetDrives())
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: will this work in non-Windows?

**Pull request merged**